source/common/upstream/upstream_impl.cc

```diff
@@ -643,6 +647,14 @@ ClusterInfoImpl::ClusterInfoImpl(
       socket_matcher_(std::move(socket_matcher)), stats_scope_(std::move(stats_scope)),
       stats_(generateStats(*stats_scope_)), load_report_stats_store_(stats_scope_->symbolTable()),
       load_report_stats_(generateLoadReportStats(load_report_stats_store_)),
+      timeout_budget_stats_store_(
+          config.track_timeout_budgets()
+              ? std::make_unique<Stats::IsolatedStoreImpl>(stats_scope_->symbolTable())
+              : nullptr),
```

Comment:
Why is this needed? I think you want to use the main scope/store for the cluster to add your stats?